### ISSUE TITLE
Evaluation

### ISSUE THREAD

**Issue by amanysh99** [autonomousvision/transfuser#321]
After training the model on all the data, the vehicle still doesn’t perform well despite many experiments! What could be the reason, even though the training visualization looked promising?
 

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/e01068a6-fac6-4ac2-acdf-055137fabb5a" />

<img width="1408" height="658" alt="Image" src="https://github.com/user-attachments/assets/9293ecbd-b4d0-4365-a8d7-943e782df00f" />
<img width="1408" height="658" alt="Image" src="https://github.com/user-attachments/assets/b82ce0ab-690c-4fe6-989b-dc3fd9f7a361" />
<img width="1408" height="658" alt="Image" src="https://github.com/user-attachments/assets/632737c3-2687-40c7-a61d-069155ae4762" />

**Comment by Kait0** [comment]
maybe the model files were not loaded correctly. You can look at how the visualization looks during inference.

Make sure there are no optimizer weight files in the folder that you are putting the model in, they can be mistakenly loaded as model.

**Comment by amanysh99** [comment]
No, the optimizer file was not loaded. The loss was 0.2. Can I calculate accuracy or something else before evaluation? I’m using the training data, which is 210GB.
The visualization while evaluation is running is not good at all:

<img width="1408" height="658" alt="Image" src="https://github.com/user-attachments/assets/f770e66d-0bf8-4cd2-ad55-bf393646a6af" />

**Comment by Kait0** [comment]
Training and validation losses are not very indicative of performance, but you can use them as sanity check.

The visualization you show suggests that the model was not loaded correctly. The depth and semantic predictions are not shown, and the bounding box predictions are random.
You can try setting [loading to strict](https://github.com/autonomousvision/transfuser/blob/3852e65e070f2efcfaba18fbbdb48057096b8a9e/team_code_transfuser/submission_agent.py#L96). Maybe that reveals some error cause.

**Comment by amanysh99** [comment]
Enabling strict=True during model loading results in an error. The cause is unclear, as the model weights appear to be compatible otherwise.

<img width="1682" height="541" alt="Image" src="https://github.com/user-attachments/assets/1bf020c2-c6b1-4688-bb2a-37f8bcb02d3c" />

**Comment by amanysh99** [comment]
Thank you, I solved it

**Comment by omkar3105** [comment]
@amanysh99 
Hi amanysh how did you solve the problem. I am getting similar problem. can you please help?
 Issue link
https://github.com/autonomousvision/transfuser/issues/329
